### PR TITLE
Perfomance fixes for quickinspector

### PR DESCRIPTION
**Pull request by krf**

### CODE REVIEW

**Comment by vkrause** [comment]
Looks good to me. Did you measure this to make sure the costs of the new code are actually lower? They should be AFAICT, but you never know ;-)

**Comment by milianw** [comment]
the sender optimization looks good to me - that call is linear in the number of connections otherwise and should now be O(1). Still, hard numbers would be good to see.

The other one is I think a bit of a misnomer - a hover event per se isn't expensive, it just happens far too often which makes it expensive to send over the wire, right? So maybe make that clear with a comment at least?

**Comment by krf** [comment]
@milianw Added a benchmark + put benchmark diff in the commit msg. Also fixed the misnomer.

Now: Trying to get my stuff compiled on MSVC2010 again...

**Comment by krf** [comment]
Fixed compilation under MSVC2010, code got slightly uglier.

File: plugins/quickinspector/quickitemmodel.cpp [hunk 1]
```diff
@@ -135,34 +135,42 @@ void QuickItemModel::populateFromItem(QQuickItem *item)
     QVector<QQuickItem *> &children = m_parentChildMap[item->parentItem()];
     std::sort(children.begin(), children.end());
 
-    // Make sure every items are known to the objects model as this is not always
-    // the case when attaching to running process (OSX)
-    Probe::instance()->discoverObject(item);
+    if (Probe::instance()) {
+        // Make sure every items are known to the objects model as this is not always
```
Comment (pasnox) [inline plugins/quickinspector/quickitemmodel.cpp]: You can remove this comments.
It clear that when attaching, one need discoverObject on any platform. is not it ?

Comment (krf) [inline plugins/quickinspector/quickitemmodel.cpp]: Could be done in another patch, yep. I'm just moving those lines here.

**Comment by krf** [comment]
Rebased. Please review.

**Pull request merged**